src/lib/nested_property.js

```diff
@@ -118,19 +120,38 @@ function npGet(cont, parts) {
  * The ONLY case we are looking for is where the entire array is selected, parts[end] === 'x'
  * AND the replacement value is an array.
  */
-function isDataArray(val, key) {
+// function isNotAContainer(key) {
+//     var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'];
 
-    var containers = ['annotations', 'shapes', 'range', 'domain', 'buttons'],
-        isNotAContainer = containers.indexOf(key) === -1;
+//     return containers.indexOf(key) === -1;
+// }
 
-    return isArray(val) && isNotAContainer;
+/*
+ * Can this value be deleted? We can delete any empty object (null, undefined, [], {})
+ * EXCEPT empty data arrays. If it's not a data array, it's a container array,
+ * ie containing objects like annotations, buttons, etc
+ */
+var DOMAIN_RANGE = /(^|.)(domain|range)$/;
+function isDeletable(val, propStr) {
```

Comment:
expanded the safe-to-delete list to include all `info_array` attributes in https://github.com/plotly/plotly.js/pull/1403/commits/1af058ab343062365eadb5ec0c60251c4c78eabf

@rreusser this also required making `args` optional for sliders and updatemenus (fallback to `[]` in command.js) since an empty array will now get pruned. Not sure there will be a real use case for this (what method could one call with no args?) but this case pops up in the tests anyhow 🏆  and it does seem more plotly-esque to not require empty parameters.